src/main/java/net/glowstone/util/GlowTravelAgent.java

```diff
@@ -0,0 +1,324 @@
+package net.glowstone.util;
+
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import net.glowstone.EventFactory;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.TravelAgent;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.world.PortalCreateEvent;
+
+@Accessors(chain = true)
+public class GlowTravelAgent implements TravelAgent {
+
+    @Getter
+    @Setter
+    private int searchRadius = 128;
+    @Getter
+    private int creationRadius = 16;
+    @Accessors(chain = false)
+    @Setter
+    private boolean canCreatePortal = true;
+    private World world;
+
+    /**
+     * Construct a new TravelAgent.
+     * @param world the world this agent operates in
+     */
+    public GlowTravelAgent(World world) {
+        if (world.getEnvironment() == World.Environment.THE_END) {
+            throw new UnsupportedOperationException("Not supported yet.");
```

Comment:
Change the message to something more descriptive, like `End portal teleportation is not supported yet.`

Reply:
Applied